### PR TITLE
1981 - Ids theme switcher conversion

### LINKED ISSUE #1981: Tests: Convert Theme Switcher Tests
**Describe the task**
Convert Theme Switcher tests from https://github.com/infor-design/enterprise-wc/tree/main/tests-to-convert/ids-theme-switcher to https://github.com/infor-design/enterprise-wc/tree/main/tests/ids-theme-switcher

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by jbrcna**
**Explain the details for making this change. What existing problem does the pull request solve?**

**Related github/jira issue (required)**:

closes #1981 

**Steps necessary to review your pull request (required)**:

Run the test suite
Check the code conversion
New converted tests should pass

**Included in this Pull Request**:
- [ ] Some documentation for the feature.
- [ ] A test for the bug or feature.
- [ ] A note to the change log.

Run the test suite
Check the code conversion
New converted tests should pass

### CODE REVIEW

**Comment by jbrcna** [comment]
I have no idea what's causing the error, I have tried it in dev tools console and it was able to distinguish and it worked.
there's no error on my end as well. usually I see a red bar beside the error.
image

**Comment by tmcconechy** [comment]
@jbrcna i found it and fixed it for you. For your info it is the lint giving the error `npm run lint`. Also suggest adding plugin: https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint

It will show errors in the code as you go

**Comment by jbrcna** [comment]
oh okay.. I didn't know that we can put it in parenthesis . I put it in <> but haven't pushed it yet because GitHub was so slow last night. thanks so much @tmcconechy .

File: tests/ids-theme-switcher/ids-theme-switcher.spec.ts [hunk 1]
```diff
@@ -63,4 +65,44 @@ test.describe('IdsThemeSwitcher tests', () => {
...
+    });
+
+    test('supports setting color variants', async () => {
+      await switcher.evaluate((idsSwitcher: IdsThemeSwitcher) => { idsSwitcher.colorVariant = 'alternate'; });
+      const colorVariant = await switcher.evaluate((idsSwitcher: IdsThemeSwitcher) => idsSwitcher!.colorVariant);
+      await expect(colorVariant).toEqual('alternate');
+    });
+
+    test('sync color variant with the container', async () => {
+      await switcher.evaluate((idsSwitcher: IdsThemeSwitcher) => { idsSwitcher.colorVariant = 'alternate'; idsSwitcher.onColorVariantRefresh(); });
```
Comment (n-ace-ancog) [inline tests/ids-theme-switcher/ids-theme-switcher.spec.ts:95]: Maybe its just my preference, but the code can be structured with the below instead for readability.

```typescript
await switcher.evaluate((idsSwitcher: IdsThemeSwitcher) => {
  idsSwitcher.colorVariant = 'alternate';
  idsSwitcher.onColorVariantRefresh();
});

const colorVariant = await switcher.evaluate((
  idsSwitcher: IdsThemeSwitcher
) => (idsSwitcher!.container as any).colorVariant);
```

File: tests/ids-theme-switcher/ids-theme-switcher.spec.ts [hunk 2]
```diff
@@ -63,4 +65,44 @@ test.describe('IdsThemeSwitcher tests', () => {
...
+        idsSwitcher?.shadowRoot?.querySelector('ids-popup-menu')?.dispatchEvent(event);
+        event.detail.elem.value = 'contrast';
+        idsSwitcher?.shadowRoot?.querySelector('ids-popup-menu')?.dispatchEvent(event);
+      });
+      const mode = await switcher.evaluate((idsSwitcher: IdsThemeSwitcher) => idsSwitcher.mode);
+      await expect(mode).toEqual('contrast');
+    });
+
+    test('can set mode and then clear it to default', async () => {
+      await switcher.evaluate((idsSwitcher: IdsThemeSwitcher) => { idsSwitcher.mode = 'dark'; idsSwitcher.mode = ''; });
```
Comment (n-ace-ancog) [inline tests/ids-theme-switcher/ids-theme-switcher.spec.ts:82]: Also, for this one instead.

```typescript
await switcher.evaluate((idsSwitcher: IdsThemeSwitcher) => {
  idsSwitcher.mode = 'dark';
  idsSwitcher.mode = '';
});
```

File: tests/ids-theme-switcher/ids-theme-switcher.spec.ts (hunk 1 above)
Comment (jbrcna) [reply tests/ids-theme-switcher/ids-theme-switcher.spec.ts:95]: sure will do. code looks nice too 👍

**Review by n-ace-ancog — COMMENTED** [review]
@jbrcna Please remove the commented lines that's causing the lint issue.

**Comment by jbrcna** [comment]
I forgot that one. will do it now in minute

**Review by n-ace-ancog — APPROVED** [review]

**Pull request merged**